src/lemmy_client.rs

```diff
@@ -43,81 +36,93 @@ impl<R: Serialize> From<R> for LemmyRequest<R> {
 mod private_trait {
   use super::{HttpType, LemmyRequest};
   use crate::errors::LemmyAppResult;
-  use async_trait::async_trait;
   use leptos::Serializable;
   use serde::{Deserialize, Serialize};
 
-  #[async_trait(?Send)]
-  pub trait LemmyClient {
+  pub trait PrivateFetch {
     async fn make_request<Response, Form, Request>(
       &self,
       method: HttpType,
       path: &str,
       form: Request,
     ) -> LemmyAppResult<Response>
     where
-      Response: Serializable + for<'de> Deserialize<'de>,
-      Form: Serialize,
+      Response: Serializable + for<'de> Deserialize<'de> + 'static,
+      Form: Serialize + core::clone::Clone + 'static + core::fmt::Debug,
       Request: Into<LemmyRequest<Form>>;
   }
 }
 
-#[async_trait(?Send)]
```

Comment:
Does rust support async traits now? That's the only reason I could see for removing this.

Reply:
yes